### DISCUSSION TITLE
Tool to manually apply a backlog?

### DISCUSSION THREAD

**Discussion by djmitche** [GothenburgBitFactory/taskwarrior#2594] [Community support]
Today, just after I'd done my weekly task triage, sync broke again, on two of my replicas.  One only had a few items in backlog.data, which I was able to apply manually after copying the task db from a still-working replica.  But the other has 50+ changes.

Does anyone have a tool that can reapply backlog items automatically?

**Comment by tbabej** [comment]
You can actually do this with plain TW! The `backlog.data` contains JSON representation of the tasks, and `task import` can handle importing and modifying (as long as the UUIDs are preserved), so the following process should work:

1. Trim the `backlog.data` from the replica with 50+ changes to just contain the new changes that you want to import (possibly the entire file).
2. Run `task import backlog_from_the_replica.data`

This will apply the modifications as specified in the backlog. See my test below for more details:
```
[warrior@tw-2.6.0 ~]$ task add one
Created task 1.

[warrior@tw-2.6.0 ~]$ task add two
Created task 2.

# Create a backup of the current state to revert to later, simulating two replicas
[warrior@tw-2.6.0 ~]$ cp ~/.task/pending.data ~/.task/old.pending.data

[warrior@tw-2.6.0 ~]$ task 1 mod pri:H
Modifying task 1 'one'.
Modified 1 task.

[warrior@tw-2.6.0 ~]$ task 2 mod pro:Work
Modifying task 2 'two'.
Modified 1 task.

[warrior@tw-2.6.0 ~]$ task 1 annotate "Some info about this task"
Annotating task 1 'one'.
Annotated 1 task.

# Revert data files back in time
[warrior@tw-2.6.0 ~]$ cp ~/.task/old.pending.data ~/.task/pending.data

[warrior@tw-2.6.0 ~]$ cp ~/.task/backlog.data recover
[warrior@tw-2.6.0 ~]$ task import recover
Importing 'recover'
 skip 96d9ea49-f96d-472f-ab22-d0d6fb20cdc0 one
 skip 21d8a05e-9513-4c15-8919-058c3a81a3c8 two
Modifying task 96d9ea49 'one'.
 mod  96d9ea49-f96d-472f-ab22-d0d6fb20cdc0 one
Modifying task 21d8a05e 'two'.
 mod  21d8a05e-9513-4c15-8919-058c3a81a3c8 two
Modifying task 96d9ea49 'one'.
 mod  96d9ea49-f96d-472f-ab22-d0d6fb20cdc0 one
Imported 5 tasks.

[warrior@tw-2.6.0 ~]$ task
[task next]

ID Age  P Project Description                            Urg 
 1 3min H         one                                     6.8
                    2021-09-04 Some info about this task     
 2 3min   Work    two                                       1

2 tasks
```

**Reply by djmitche** [reply]
Oh, fantastic -- thank you!
